pkg/reconcile/pipeline/handler/collect/impl.go

```diff
@@ -131,6 +132,27 @@ func ProvisionedService(ctx pipeline.Context) {
 	}
 }
 
+func DirectSecretReference(ctx pipeline.Context) {
+	services, err := ctx.Services()
+	if err != nil {
+		requestRetry(ctx, ErrorReadingServicesReason, err)
+		return
+	}
```

Comment:
You can omit this here, because we have this check already in PreFlight stage:

https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/reconcile/pipeline/handler/collect/impl.go#L29

This stage was created to perform common checks for all followup stages.

Reply:
Done. 10cf41199e4e0a45796f32d33108c79969cac1f7